File: spec/reducers/view.spec.js
import chai from "chai";
import V    from "../../src/reducers/view";
import PV   from "../../src/constants/PanelView";
import AT   from "../../src/constants/ActionTypes";

chai.should();

describe("view reducer", () => {

  it("should be a function", () => {
    V.should.be.a("function");
  });

  describe("right", () => {
    it("is a key in view store", () => {
      V().should.have.ownProperty("right");
    });
    it("should change to IMPRINT view after action SHOW_IMPRINT", () => {
      V({}, {
        type: AT.SHOW_IMPRINT
      }).right.should.equal(PV.IMPRINT);
    });
  });

  describe("left", () => {
    it("should change to RESULT view after action SEARCH_RESULT", () => {
      V({}, {
        type: AT.SEARCH_RESULT
      }).left.should.equal(PV.RESULT);
    });
    it("should not be changed after action SEARCH_RESULT if it exists", () => {
      V({
        left: "foo"
      }, {
        type: AT.SEARCH_RESULT
      }).left.should.equal("foo");
    });
  });

  describe("the menu", () => {
    // it("should be active in the initial state", () => {
    //   V().menu.should.equal(true);
    // });
    // it("visibility can be toggled", () => {
    //   const state1 = V(V(), {
    //     type: AT.TOGGLE_MENU
    //   });
    //   state1.menu.should.equal(false);
    //   const state2 = V(state1, {
    //     type: AT.TOGGLE_MENU
    //   });
    //   state2.menu.should.equal(true);
    // });
    it("cannot be opend if a modal dialog is open", () => {
      const state = V({
        modal: PV.LOCATE,
        menu: false
      }, {
        type: AT.TOGGLE_MENU
      });
      state.menu.should.equal(false);
    });
  });
});

File: spec/reducers/view.spec.rs
// TODO: import chai from "chai";
// TODO: import V    from "../../src/reducers/view";
// TODO: import PV   from "../../src/constants/PanelView";
// TODO: import AT   from "../../src/constants/ActionTypes";
// TODO: 
// TODO: chai.should();
// TODO: 
// TODO: describe("view reducer", () => {
// TODO: 
// TODO:   it("should be a function", () => {
// TODO:     V.should.be.a("function");
// TODO:   });
// TODO: 
// TODO:   describe("right", () => {
// TODO:     it("is a key in view store", () => {
// TODO:       V().should.have.ownProperty("right");
// TODO:     });
// TODO:     it("should change to IMPRINT view after action SHOW_IMPRINT", () => {
// TODO:       V({}, {
// TODO:         type: AT.SHOW_IMPRINT
// TODO:       }).right.should.equal(PV.IMPRINT);
// TODO:     });
// TODO:   });
// TODO: 
// TODO:   describe("left", () => {
// TODO:     it("should change to RESULT view after action SEARCH_RESULT_ENTRIES", () => {
// TODO:       V({}, {
// TODO:         type: AT.SEARCH_RESULT_ENTRIES
// TODO:       }).left.should.equal(PV.RESULT);
// TODO:     });
// TODO:     it("should not be changed after action SEARCH_RESULT_ENTRIES if it exists", () => {
// TODO:       V({
// TODO:         left: "foo"
// TODO:       }, {
// TODO:         type: AT.SEARCH_RESULT_ENTRIES
// TODO:       }).left.should.equal("foo");
// TODO:     });
// TODO:   });
// TODO: 
// TODO:   describe("the menu", () => {
// TODO:     // it("should be active in the initial state", () => {
// TODO:     //   V().menu.should.equal(true);
// TODO:     // });
// TODO:     // it("visibility can be toggled", () => {
// TODO:     //   const state1 = V(V(), {
// TODO:     //     type: AT.TOGGLE_MENU
// TODO:     //   });
// TODO:     //   state1.menu.should.equal(false);
// TODO:     //   const state2 = V(state1, {
// TODO:     //     type: AT.TOGGLE_MENU
// TODO:     //   });
// TODO:     //   state2.menu.should.equal(true);
// TODO:     // });
// TODO:     it("cannot be opend if a modal dialog is open", () => {
// TODO:       const state = V({
// TODO:         modal: PV.LOCATE,
// TODO:         menu: false
// TODO:       }, {
// TODO:         type: AT.TOGGLE_MENU
// TODO:       });
// TODO:       state.menu.should.equal(false);
// TODO:     });
// TODO:   });
// TODO: });
